libarchive/archive_read_support_format_tar.c

```diff
@@ -2302,13 +2295,15 @@ pax_attribute(struct archive_read *a, struct tar *tar, struct archive_entry *ent
 						err = ARCHIVE_FAILED;
 					} else {
 						p = __archive_read_ahead(a, value_length, &bytes_read);
-						if (p != NULL) {
-							if (gnu_sparse_01_parse(a, tar, p, value_length) != ARCHIVE_OK) {
-								err = ARCHIVE_WARN;
-							}
-						} else {
+						if (p == NULL) {
```

Comment:
I've swapped this and a few other instances around so that we consistently check for `p == NULL` as the first thing after `__archive_read_ahead`.